cpp/src/groupby/hash/compute_aggregations.cuh

```diff
@@ -67,8 +67,10 @@ rmm::device_uvector<cudf::size_type> compute_aggregations(
   auto const grid_size =
     max_occupancy_grid_size<typename SetType::ref_type<cuco::insert_and_find_tag>>(num_rows);
   auto const available_shmem_size = get_available_shared_memory_size(grid_size);
+  auto const offsets_buffer_size  = compute_shmem_offsets_size(flattened_values.num_columns()) * 2;
+  auto const data_buffer_size     = available_shmem_size - offsets_buffer_size;
   auto const has_sufficient_shmem =
-    available_shmem_size > (compute_shmem_offsets_size(flattened_values.num_columns()) * 2);
+    (data_buffer_size > 0) and (data_buffer_size < (GROUPBY_CARDINALITY_THRESHOLD * 8));
```

Comment:
If I'm reading this right, the logic is the same as before, up to `and (data_buffer_size < (GROUPBY_CARDINALITY_THRESHOLD * 8))`. Does that mean we are skipping shmem groupby if we have _too much_ leftover shared memory? Happy to pick up the conversation again today if you want to talk through it.

Reply:
Good catch! That was a bug, but I just pushed the fix, so it should be all set now.